src/filename.rs

```diff
@@ -76,3 +81,13 @@ fn extension_is_still_in_basename() {
 fn domain_prefix_www_doesnt_matter() {
     assert_eq!(tb("https://edjopato.de/"), tb("https://www.edjopato.de/"));
 }
+
+#[test]
```

Comment:
Would you mind adding these tests:
`http://127.0.0.1:12345/test/` and `http://[fe80::1234%eth0]/test/`
The last one is not a valid URI, BUT it contains a valid scoped IPv6. We should expect people to do/use/test this hence we should have a test for it. Even if the tests says "if scoped, throw error".

Reply:
I added both. The second one fails as expected.
I also added the port to the base name when it is specified in the url.